### PR TITLE
Fix broken link in releasing instructions.

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Documentation

### Description

Fixing broken link in release instructions.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -11,11 +11,18 @@ In addition to the build pre-requistes:
...
 the release script to actually perform the release and prepare main for the next development version.
 
 ```shell
 ./scripts/release.sh -k <YOUR_KEY_SHORT_ID> -v <RELEASE_VERSION>
 ```
 
+where `<YOUR_KEY_SHORT_ID>` is the short id of your PGP key
+and `<RELEASE_VERSION>` is a release number such as `0.6.0`.
+
+Once the release is completed, use the [Nexus UI](https://s01.oss.sonatype.org/) to close the staging repository, then release it. That will push the Maven artefacts available
```
Comment (SamBarker) [inline RELEASING.md:24]: We should add a note about the timing of merging the version bump. Closing the staging repository involves validation which has the potential to fail and thus need fixing...
Options:

1. Don't merge until the staging repository is closed. 
2. Lazy branching on errors closing the staging repository and re-running the release from there.

**Review by SamBarker — APPROVED** [review]
LGTM

File: RELEASING.md (hunk 1 above)
Comment (k-wall) [reply RELEASING.md:24]: I'll merge this PR and we can handle this comment in a new one.

Comment (k-wall) [reply RELEASING.md:24]: I opened PR #202

**Pull request merged**